src/ansys/units/unit.py

```diff
@@ -391,16 +391,18 @@ def _units_to_dim(
                 dimensions[BaseDimensions[idx]] += unit_term_exponent
             else:
                 dimensions[BaseDimensions[idx]] = unit_term_exponent
-
         # Retrieve derived unit composition unit string and SI factor.
-        if unit_term in _derived_units:
+        elif unit_term in _derived_units:
             # Recursively parse composition unit string
 
             dimensions = _units_to_dim(
                 units=_derived_units[unit_term]["composition"],
                 exponent=unit_term_exponent,
                 dimensions=dimensions,
             )
+        else:
+            if _:
+                raise UnimplementedUnit(_)
```

Comment:
why not elif?

Reply:
`_` is empty for implemented unit without multiplier. 

```
    qt = Quantity(5.0, "m^0")
    qtm = qt**2
```

and if we try to execute  `qtm = qt**2` we get error so to avoid this we have added it in else block.

![image](https://github.com/ansys/pyansys-units/assets/106588300/fec25ee2-c94c-4c6b-853c-9834d6d4e4f8)

Reply:
@hpohekar 

Sorry, to clarify: all I meant was that we have something like:
```
if x: 
  ...
elif y:
  ... 
else:
  if z:
    ...
```
which is equivalent to:
```
if x: 
  ...
elif y:
  ... 
elif z:
  ...
```

Reply:
@seanpearsonuk  Right, got it. I added it for debugging but I haven't updated it later. Done. Thank you.